src/wp-includes/class-wp-list-util.php

```diff
@@ -166,9 +166,17 @@ public function pluck( $field, $index_key = null ) {
 			foreach ( $this->output as $key => $value ) {
 				if ( is_object( $value ) ) {
 					$newlist[ $key ] = $value->$field;
-				} else {
+				} elseif ( is_array( $value ) ) {
 					$newlist[ $key ] = $value[ $field ];
 				}
+
+				if ( ! ( is_object( $value ) || is_array( $value ) ) ) {
+					_doing_it_wrong(
+						__FUNCTION__,
+						__( 'Value for `$this->output` must be either an object or an array for this function.' ),
```

Comment:
```suggestion
						__( "Values for the input array must be either objects or arrays.' ),
```
Same suggestion in the later instance.